### PR TITLE
storage: Export mount points etc for inactive logical volumes

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

File: pkg/storaged/anaconda.jsx [hunk 1]
```diff
@@ -30,19 +30,38 @@ export const AnacondaAdvice = () => {
     return null;
 };
 
+function parse_parent_from_options(options) {
+    const parent_match = options.match(/x-parent=(?<parent>[\w\\-]+)/);
+    if (parent_match) {
+        console.log("PARENT", parent_match.groups.parent);
+        return parent_match.groups.parent;
+    } else
+        return null;
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:39]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19951-20240206-142939-b945d329-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx [hunk 2]
```diff
@@ -96,7 +134,21 @@
...
     for (const p in client.blocks) {
         const b = client.blocks[p];
-        mpm[decode_filename(b.Device)] = block_info(b);
+        mpm[decode_filename(b.PreferredDevice)] = block_info(b);
+    }
+
+    // Add inactive logical volumes
+    for (const vg_p in client.vgroups) {
+        const vg = client.vgroups[vg_p];
+        for (const lv of client.vgroups_lvols[vg_p] || []) {
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:143]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19951-20240206-142939-b945d329-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx [hunk 3]
```diff
@@ -30,19 +30,41 @@ export const AnacondaAdvice = () => {
     return null;
 };
 
+function parse_parent_from_options(options) {
+    const parent_match = options.match(/x-parent=(?<parent>[\w\\-]+)/);
+    if (parent_match) {
+        return parent_match.groups.parent;
+    } else
+        return null;
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:38]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19951-20240207-125203-ae8dc098-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx [hunk 4]
```diff
@@ -96,7 +137,21 @@
...
     for (const p in client.blocks) {
         const b = client.blocks[p];
-        mpm[decode_filename(b.Device)] = block_info(b);
+        mpm[device_name(b)] = block_info(b);
+    }
+
+    // Add inactive logical volumes
+    for (const vg_p in client.vgroups) {
+        const vg = client.vgroups[vg_p];
+        for (const lv of client.vgroups_lvols[vg_p] || []) {
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:146]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19951-20240207-125203-ae8dc098-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx [hunk 5]
```diff
@@ -30,19 +30,41 @@ export const AnacondaAdvice = () => {
...
+function parse_parent_from_options(options) {
+    const parent_match = options.match(/x-parent=(?<parent>[\w\\-]+)/);
+    if (parent_match) {
+        return parent_match.groups.parent;
+    } else
+        return null;
+}
+
+function uuid_equal(a, b) {
+    return a == b; // XXX
```
Comment (mvollmer) [inline pkg/storaged/anaconda.jsx:42]: This needs to be addressed.

File: pkg/storaged/anaconda.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:38]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19951-20240208-075016-3fea3d1d-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx (hunk 4 above)
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:146]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19951-20240208-075016-3fea3d1d-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:38]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19951-20240208-091711-3fea3d1d-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx (hunk 4 above)
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:146]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19951-20240208-091711-3fea3d1d-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by KKoukiou — APPROVED** [review]
This might will obsolete when re-scaning gets fixed in the backend - but we need it for now.

**Comment by mvollmer** [comment]
> This might will obsolete when re-scaning gets fixed in the backend - but we need it for now.

Well, deactivated logical volumes can happen for other reasons than a blivet re-scan, and it is good to be able to deal with them in any case.

File: pkg/storaged/anaconda.jsx [hunk 6]
```diff
@@ -22,19 +22,41 @@ import client from "./client.js";
 import { decode_filename } from "./utils.js";
 import { parse_subvol_from_options } from "./btrfs/utils.jsx";
 
+function parse_parent_from_options(options) {
+    const parent_match = options.match(/x-parent=(?<parent>[\w\\-]+)/);
+    if (parent_match) {
+        return parent_match.groups.parent;
+    } else
+        return null;
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:30]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19951-20240209-081459-7aeeff60-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/anaconda.jsx [hunk 7]
```diff
@@ -88,7 +129,21 @@
...
     for (const p in client.blocks) {
         const b = client.blocks[p];
-        mpm[decode_filename(b.Device)] = block_info(b);
+        mpm[device_name(b)] = block_info(b);
+    }
+
+    // Add inactive logical volumes
+    for (const vg_p in client.vgroups) {
+        const vg = client.vgroups[vg_p];
+        for (const lv of client.vgroups_lvols[vg_p] || []) {
```
Comment (cockpituous) [inline pkg/storaged/anaconda.jsx:138]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19951-20240209-081459-7aeeff60-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**